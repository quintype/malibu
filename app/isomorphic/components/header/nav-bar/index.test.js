import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import { NavBar } from ".";

const renderer = new ShallowRenderer();

let container;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

describe("Navbar", () => {
  const defaultMenu = {
    default: []
  };

  const menuData = {
    default: [
      {
        "updated-at": 1604300936817,
        "tag-name": null,
        "entity-properties": null,
        "entity-slug": null,
        "publisher-id": 15,
        "menu-group-slug": "default",
        "item-id": 23224,
        rank: 38,
        title: "For Developers",
        "item-type": "section",
        "section-slug": "for-developers",
        "tag-slug": null,
        id: 2243,
        "parent-id": null,
        url: "https://madrid-old.quintype.io/for-developers",
        "entity-name": null,
        "created-at": 1544609755177,
        "section-name": "For Developers",
        "menu-group-id": 95,
        data: '{color: "#000000"}',
        children: "[]",
        completeUrl: "/for-developers"
      },
      {
        "updated-at": 1604300936818,
        "tag-name": null,
        "entity-properties": null,
        "entity-slug": null,
        "publisher-id": 15,
        "menu-group-slug": "default",
        "item-id": 6392,
        rank: 39,
        title: "Product",
        "item-type": "section",
        "section-slug": "mobile",
        "tag-slug": null,
        id: 2240,
        "parent-id": null,
        url: "https://madrid-old.quintype.io/tech/mobile",
        "entity-name": null,
        "created-at": 1544608844876,
        "section-name": "Mobile",
        "menu-group-id": 95,
        data: '{color: "#000000"}',
        children: "[{â€¦}, {â€¦}]",
        completeUrl: "/tech/mobile"
      }
    ]
  };

  it("renders the navbar with default vaules", () => {
    const tree = renderer.render(<NavBar />);
    expect(tree).toMatchSnapshot();
  });

  it("renders the navbar when menu.default is empty array", () => {
    const tree = renderer.render(<NavBar menu={defaultMenu} />);
    expect(tree).toMatchSnapshot();
  });

  it("renders the navbar when menu.default is a valid array", () => {
    const tree = renderer.render(<NavBar menu={menuData} />);
    expect(tree).toMatchSnapshot();
  });
});
