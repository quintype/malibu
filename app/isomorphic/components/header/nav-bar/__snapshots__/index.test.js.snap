// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders the navbar when menu.default is a valid array 1`] = `
<React.Fragment>
  <AppLogo />
  <ul
    className="styles-m__navbar__1m8_H"
  >
    <li
      className="styles-m__menu-item__1szXn styles-m__desktop-view__1ts3k"
    >
      <MenuItem
        item={
          Object {
            "children": "[]",
            "completeUrl": "/for-developers",
            "created-at": 1544609755177,
            "data": "{color: \\"#000000\\"}",
            "entity-name": null,
            "entity-properties": null,
            "entity-slug": null,
            "id": 2243,
            "item-id": 23224,
            "item-type": "section",
            "menu-group-id": 95,
            "menu-group-slug": "default",
            "parent-id": null,
            "publisher-id": 15,
            "rank": 38,
            "section-name": "For Developers",
            "section-slug": "for-developers",
            "tag-name": null,
            "tag-slug": null,
            "title": "For Developers",
            "updated-at": 1604300936817,
            "url": "https://madrid-old.quintype.io/for-developers",
          }
        }
      />
    </li>
    <li
      className="styles-m__menu-item__1szXn styles-m__desktop-view__1ts3k"
    >
      <MenuItem
        item={
          Object {
            "children": "[{…}, {…}]",
            "completeUrl": "/tech/mobile",
            "created-at": 1544608844876,
            "data": "{color: \\"#000000\\"}",
            "entity-name": null,
            "entity-properties": null,
            "entity-slug": null,
            "id": 2240,
            "item-id": 6392,
            "item-type": "section",
            "menu-group-id": 95,
            "menu-group-slug": "default",
            "parent-id": null,
            "publisher-id": 15,
            "rank": 39,
            "section-name": "Mobile",
            "section-slug": "mobile",
            "tag-name": null,
            "tag-slug": null,
            "title": "Product",
            "updated-at": 1604300936818,
            "url": "https://madrid-old.quintype.io/tech/mobile",
          }
        }
      />
    </li>
    <li>
      <button
        onClick={[Function]}
      >
        Login
      </button>
    </li>
  </ul>
  <NavbarSearch />
</React.Fragment>
`;

exports[`Navbar renders the navbar when menu.default is empty array 1`] = `
<React.Fragment>
  <AppLogo />
  <ul
    className="styles-m__navbar__1m8_H"
  >
    <li>
      <button
        onClick={[Function]}
      >
        Login
      </button>
    </li>
  </ul>
  <NavbarSearch />
</React.Fragment>
`;

exports[`Navbar renders the navbar with default vaules 1`] = `
<React.Fragment>
  <AppLogo />
  <ul
    className="styles-m__navbar__1m8_H"
  >
    <li>
      <button
        onClick={[Function]}
      >
        Login
      </button>
    </li>
  </ul>
  <NavbarSearch />
</React.Fragment>
`;
